lmfit/model.py

```diff
@@ -1116,9 +1116,10 @@ def __init__(self, left, right, op, **kws):
                         "use distinct names.")
             raise NameError(msg)
 
-        # we assume that all the sub-models have the same independent vars
+        # every independent_vars of the left and right model is added to independent_vars of the CompositeModel
         if 'independent_vars' not in kws:
-            kws['independent_vars'] = self.left.independent_vars
+            ivar_left, ivar_right = self.left.independent_vars, self.right.independent_vars
+            kws['independent_vars'] = ivar_left + [ivar for ivar in ivar_right if ivar not in ivar_left]
```

Comment:
how about 
```kws['independent_vars'] = np.unique(self.left.independent_vars + self.right.independent_vars)```
?

Reply:
sorry, this was my bad... it should return a `list` not an `array` so that conversion is required here